vcstool/commands/import_.py

```diff
@@ -33,12 +38,20 @@ def __init__(self, args, url, version=None, recursive=False):
         self.recursive = recursive
 
 
+def file_or_url_type(x):
+    try:
+        return urlopen(x)
+    except ValueError:
+        return argparse.FileType('r')(x)
+
+
 def get_parser():
     parser = argparse.ArgumentParser(
         description='Import the list of repositories', prog='vcs import')
     group = parser.add_argument_group('"import" command parameters')
     group.add_argument(
-        '--input', type=argparse.FileType('r'), default=sys.stdin)
+        '--input', type=file_or_url_type, default='-',
```

Comment:
The loading of the URL should happen and arg parsing since it could take an arbitrary amount of time. Considering the case that e.g. completion is being computed using `parse_known_args`, Please move it loading of the URL post arg parsing.

Reply:
Unrelated question: why does the patch change the `default` argument?

Reply:
Because argparse formats it weird otherwise. `-` is another name for stdin.